features/FEATURE_COMMON_PAL/mbed-trace/source/mbed_trace.c

```diff
@@ -153,6 +161,7 @@ static trace_t m_trace = {
 
 int mbed_trace_init(void)
 {
+#ifndef MBED_TRACE_STATIC
```

Comment:
Also could fill this in the structure or in the init method (or another way) if it was preferred?